### ISSUE TITLE
Missing `app_mention` in the model

### ISSUE THREAD

**Issue by medwards** [abdolence/slack-morphism-rust#23]
I'm experimenting /w the example and added [app_mention](https://api.slack.com/events/app_mention) bot event. When I @ my bot (`echo-bot`) I get the following error

```
SlackClientProtocolError {
    json_error: Error("unknown variant `app_mention`, expected `message` or `app_home_opened`", line: 0, column: 0),
    http_response_body: "{\"token\":\"fYGMuMwItDR0DHT70bXofHoW\",\"team_id\":\"T01RJKRAW3A\",\"api_app_id\":\"A01R68WN84U\",\"event\":{\"client_msg_id\":\"22f15c93-03fb-4b0a-946f-88944e18d853\",\"type\":\"app_mention\",\"text\":\"<@U01R0BG5K6X> I mentioned echo bot\",\"user\":\"U01R7DVQXCH\",\"ts\":\"1615501392.000800\",\"team\":\"T01RJKRAW3A\",\"blocks\":[{\"type\":\"rich_text\",\"block_id\":\"eduC\",\"elements\":[{\"type\":\"rich_text_section\",\"elements\":[{\"type\":\"user\",\"user_id\":\"U01R0BG5K6X\"},{\"type\":\"text\",\"text\":\" I mentioned echo bot\"}]}]}],\"channel\":\"C01QV1E7TCK\",\"event_ts\":\"1615501392.000800\"},\"type\":\"event_callback\",\"event_id\":\"Ev01RQ3W7XLG\",\"event_time\":1615501392,\"authorizations\":[{\"enterprise_id\":null,\"team_id\":\"T01RJKRAW3A\",\"user_id\":\"U01R0BG5K6X\",\"is_bot\":true,\"is_enterprise_install\":false}],\"is_ext_shared_channel\":false,\"event_context\":\"1-app_mention-T01RJKRAW3A-C01QV1E7TCK\"}",
}
```

Is it because we need to fill out the [SlackEventCallbackBody](https://docs.rs/slack-morphism/0.5.5/slack_morphism/prelude/events/enum.SlackEventCallbackBody.html) enum?

If this is the case it puts subscribers at risk of being throttled and disabled. [Slack will disable bots that fail more than 5% of the time.](https://api.slack.com/apis/connections/events-api#failure_limits).

**Comment by medwards** [comment]
Another example of a different but very similar error:
```
SlackClientProtocolError {
    json_error: Error("unknown variant `message_changed`, expected one of `bot_message`, `me_message`, `channel_join`, `bot_add`, `bot_remove`, `channel_topic`, `channel_purpose`, `channel_name`", line: 0, column: 0),
    http_response_body: "{\"token\":\"fYGMuMwItDR0DHT70bXofHoW\",\"team_id\":\"T01RJKRAW3A\",\"api_app_id\":\"A01R68WN84U\",\"event\":{\"type\":\"message\",\"subtype\":\"message_changed\",\"hidden\":true,\"message\":{\"client_msg_id\":\"c9400470-c9e7-4aac-8482-c2d8433166e6\",\"type\":\"message\",\"text\":\"echo bot working, code here <https:\\/\\/github.com\\/medwards\\/slack_echo_bot>\",\"user\":\"U01R7DVQXCH\",\"team\":\"T01RJKRAW3A\",\"attachments\":[{\"service_name\":\"GitHub\",\"title\":\"medwards\\/slack_echo_bot\",\"title_link\":\"https:\\/\\/github.com\\/medwards\\/slack_echo_bot\",\"text\":\"Experimenting \\/w slack-morphism-rust. Contribute to medwards\\/slack_echo_bot development by creating an account on GitHub.\",\"fallback\":\"GitHub: medwards\\/slack_echo_bot\",\"thumb_url\":\"https:\\/\\/avatars.githubusercontent.com\\/u\\/7339?s=400&v=4\",\"from_url\":\"https:\\/\\/github.com\\/medwards\\/slack_echo_bot\",\"thumb_width\":400,\"thumb_height\":400,\"service_icon\":\"https:\\/\\/a.slack-edge.com\\/80588\\/img\\/unfurl_icons\\/github.png\",\"id\":1,\"original_url\":\"https:\\/\\/github.com\\/medwards\\/slack_echo_bot\"}],\"blocks\":[{\"type\":\"rich_text\",\"block_id\":\"iOi\",\"elements\":[{\"type\":\"rich_text_section\",\"elements\":[{\"type\":\"text\",\"text\":\"echo bot working, code here \"},{\"type\":\"link\",\"url\":\"https:\\/\\/github.com\\/medwards\\/slack_echo_bot\"}]}]}],\"ts\":\"1615562442.000200\"},\"channel\":\"C01QV1E7TCK\",\"previous_message\":{\"client_msg_id\":\"c9400470-c9e7-4aac-8482-c2d8433166e6\",\"type\":\"message\",\"text\":\"echo bot working, code here <https:\\/\\/github.com\\/medwards\\/slack_echo_bot>\",\"user\":\"U01R7DVQXCH\",\"ts\":\"1615562442.000200\",\"team\":\"T01RJKRAW3A\",\"blocks\":[{\"type\":\"rich_text\",\"block_id\":\"iOi\",\"elements\":[{\"type\":\"rich_text_section\",\"elements\":[{\"type\":\"text\",\"text\":\"echo bot working, code here \"},{\"type\":\"link\",\"url\":\"https:\\/\\/github.com\\/medwards\\/slack_echo_bot\"}]}]}]},\"event_ts\":\"1615562444.000400\",\"ts\":\"1615562444.000400\",\"channel_type\":\"channel\"},\"type\":\"event_callback\",\"event_id\":\"Ev01R31SPA74\",\"event_time\":1615562444,\"authorizations\":[{\"enterprise_id\":null,\"team_id\":\"T01RJKRAW3A\",\"user_id\":\"U01R0BG5K6X\",\"is_bot\":true,\"is_enterprise_install\":false}],\"is_ext_shared_channel\":false,\"event_context\":\"1-message-T01RJKRAW3A-C01QV1E7TCK\"}",
}
```

**Comment by abdolence** [comment]
Yes, it has still limited models for events support in
`SlackEventCallbackBody`. 

It is easy to introduce all required events, just need more models, so PR's are welcome :)
I'll update some soon anyway myself.

**Comment by abdolence** [comment]
Updated mentioned types here:
https://github.com/abdolence/slack-morphism-rust/commit/c1e2455b377d87942685250006e35015718cda9f

**Comment by abdolence** [comment]
Should be fixed now in v0.6.0.
